timApp/termsofservice/tos_acceptance.py

```diff
@@ -0,0 +1,36 @@
+from flask import Response, current_app
+from datetime import datetime
+
+from timApp.auth.sessioninfo import get_current_user_object
+from timApp.item.item import Item
+from timApp.timdb.sqa import db
+from timApp.util.flask.responsehelper import ok_response
+from timApp.util.flask.typedblueprint import TypedBlueprint
+from timApp.util.utils import get_current_time
+
+tos_accepted = TypedBlueprint("tos_accepted", __name__, url_prefix="")
+
+
+@tos_accepted.post("/tosagree")
+def tosagree() -> Response:
+    curr_user = get_current_user_object()
+
+    if curr_user.is_real_user:
+        now = get_current_time()
+        if curr_user.tos_accepted_at != now:
+            curr_user.tos_accepted_at = now
+            db.session.commit()
+    return ok_response()
+
+
+def get_tos_date() -> datetime | None:
```

Comment:
Tähän kohtaan ehdotan vielä laittamaan jonkun yksinkertaisen cachen, koska tämän dokumentin tietoja ei tarvitse oikeastaan hakea koko ajan: jotain mallia:

```py
from functools import cache

@cache
def get_tos_date() -> datetime | None:
```

tällöin tulos lasketaan vain kerran jokaiselle workerille kunnes workeri käynnistetään uudelleen. Se on sinänsä ihan toimiva ratkaisu tähän.